### PR TITLE
sysdiagnose: CLI timeout implementation and callback to track progress

### PR DESCRIPTION
**Pull request by dario-br**
1. crash CLI now implements timeout parameter to generate sysdiagnose file
2. crash_reports: get_new_sysdiagnose accepts a callback to track progress of the sysdiagnose generation. The callback simply admits one parameter with the elapsed time so far.

### CODE REVIEW

File: pymobiledevice3/services/crash_reports.py [hunk 1]
```diff
@@ -129,22 +129,36 @@ def watch(self, name: str = None, raw: bool = False) -> Generator[str, None, Non
                 else:
                     yield crash_report
 
-    def get_new_sysdiagnose(self, out: str, erase: bool = True, *, timeout: Optional[float] = None) -> None:
+    def get_new_sysdiagnose(self, out: str, erase: bool = True, *, timeout: Optional[float] = None,
```
Comment (doronz88) [inline pymobiledevice3/services/crash_reports.py:132]: add exact typing to `callback`

**Review by doronz88 — CHANGES_REQUESTED** [review]
Thanks for the PR!
I'm currently trying to make sure more parts of pymobile are typed accordingly, so please add the typing to new parameters

File: pymobiledevice3/services/crash_reports.py (hunk 1 above)
Comment (dario-br) [reply pymobiledevice3/services/crash_reports.py:132]: Noted! let me know if the commit below works!

**Review by doronz88 — APPROVED** [review]

**Comment by doronz88** [comment]
Thanks for the PR! 😃

**Pull request merged**